server/ipqueue.go

```diff
@@ -207,6 +207,43 @@ func (q *ipQueue[T]) popOne() (T, bool) {
 	return e, true
 }
 
+// Returns the last element from the queue, if any. See comment above
+// regarding calling after being notified that there is something and
+// the use of drain(). In short, the caller should always check the
+// boolean return value to ensure that the value is genuine and not a
+// default empty value.
+func (q *ipQueue[T]) popOneLast() (T, bool) {
```

Comment:
nit: `popLast()` may be enough.

Adding a test for this new API would be a good thing, for instance making sure that we can do popOne() and popLast() mixed for instance.